ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java

```diff
@@ -372,7 +372,11 @@ public void onProgress(long bytesWritten, long contentLength, boolean done) {
           return;
         }
       } else {
-        requestBody = RequestBody.create(contentMediaType, body.getBytes(StandardCharsets.UTF_8));
+        // Use getBytes() to convert the body into a byte[], preventing okhttp from
+        // appending the character set to the Content-Type header when otherwise unspecified
+        // https://github.com/facebook/react-native/issues/8237
+        Charset charset = contentMediaType.charset(StandardCharsets.UTF_8);
```

Comment:
Any chance for NPE if contentMediaType is null?

Reply:
Yes:

https://github.com/square/okhttp/blob/19466ca4c0d43ebc69cdbc219b0298a8a6db30b4/okhttp/src/main/java/okhttp3/MediaType.java#L104-L110

I'll make this change, :+1: